cmd/dagger/download.go

```diff
@@ -3,37 +3,34 @@ package main
 import (
 	"fmt"
 
-	"dagger.io/dagger"
 	"github.com/spf13/cobra"
 )
 
 var exportPath string
 
 var downloadCmd = &FuncCommand{
 	Name:  "download",
-	Short: "Download an asset to the host (directory, file, container).",
-	OnInit: func(cmd *cobra.Command) {
-		cmd.PersistentFlags().StringVar(&exportPath, "export-path", "", "Path to export to")
-		cmd.MarkFlagRequired("export-path")
+	Short: "Download an asset from module function",
+	Long:  "Download an asset returned by a module function and save it to the host.\n\nWorks with a Directory, File or Container.",
+	Init: func(cmd *cobra.Command) {
+		cmd.PersistentFlags().StringVar(&exportPath, "export-path", ".", "Path to export to")
 	},
-	OnSelectObject: func(c *callContext, _ string) (*modTypeDef, error) {
-		c.Select("export")
-		c.Arg("path", exportPath)
-		return nil, nil
+	OnSelectObjectLeaf: func(c *FuncCommand, name string) error {
+		switch name {
+		case Directory, File, Container:
```

Comment:
What if we detect if the path is to an existing dir and act accordingly?

Reply:
> What if we detect if the path is to an existing dir and act accordingly?

That's what that API does for you; if it is an existing dir, it will export the file to be inside it; if the path is an existing file, it will overwrite it; if the path doesn't exist then it will just write the file there.

So I think we should be able to just rely on the API rather than re-implement that.

Reply:
Oh, so the allowParentDirPath just triggers that behavior, so we can set it always to true in the CLI, is that right?

Reply:
Yep exactly, I actually added that option to the API in one of the various previous iterations of `dagger do` to help this exact situation (besides that it's somewhat useful in general too).

Reply:
Done